File: Android/app/src/main/java/ru/inwords/inwords/domain/interactor/authorisation/AuthorisationInteractor.java
package ru.inwords.inwords.domain.interactor.authorisation;

import io.reactivex.Completable;
import ru.inwords.inwords.data.dto.UserCredentials;

public interface AuthorisationInteractor {
    Completable signIn(UserCredentials userCredentials);

    Completable signUp(UserCredentials userCredentials);
}

File: Android/app/src/main/java/ru/inwords/inwords/domain/interactor/authorisation/AuthorisationInteractor.kt
package ru.inwords.inwords.domain.interactor.authorisation

import io.reactivex.Completable
import ru.inwords.inwords.profile.data.bean.UserCredentials

interface AuthorisationInteractor {
    fun trySignInExistingAccount(): Completable

    fun signIn(userCredentials: UserCredentials): Completable

    fun signUp(userCredentials: UserCredentials): Completable
}
